ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -262,11 +266,25 @@ impl<'a> Assemble<'a> {
             // is x86 only, we don't need to worry about portability.
             #[allow(clippy::fn_to_numeric_cast)]
             {
+                dynasm!(self.asm; => deopt_label);
+                // Push all the general purpose registers to the stack.
+                for (i, reg) in lsregalloc::GP_REGS.iter().rev().enumerate() {
+                    if *reg == Rq::RSI {
+                        // RSI is handled differently in `fail_label`: RSI now contains the deopt
+                        // ID, so we have to fish the actual value out of the stack. See the FIXME
+                        // in `fail_label`.
+                        let off = i32::try_from(i * 8).unwrap();
+                        dynasm!(self.asm; push QWORD [rsp + off]);
+                    } else {
+                        dynasm!(self.asm; push Rq(reg.code()));
+                    }
+                }
                 dynasm!(self.asm
-                    ;=> deopt_label
                     ; mov rdi, [rbp]
                     ; mov rdx, rbp
+                    ; mov rcx, rsp
```

Comment:
Now that is something I had not thought of doing.

Reply:
I am not sure if I detect admiration or disgust :p

Reply:
I was undecided at first, but I think I have settled on admiration. I would probably have used a `naked` function to pull the registers back out from the stack, but turning `rsp` into `[u64; 16]` is very clever.

Reply:
I just did the dumbest thing I could think of. I'm pretty sure you can come up with a better / faster approach (e.g. passing some values in registers rather than spilling), but I figured that we (probably you, because you're better at this than me) could do this in a later PR.